File: Pattern Printing/Python/Diamond-pyramid.py

# Diamond Pyramid
rows = int(input("Enter the number of of rows you want in a pyramid (will also give a dropdown pyramid attached to it):"))
back_col = rows - 1 #After experiments, this step was really importat to do back_col = rows - 1 instead of directly
                    #doing back_col = 4
space = rows 
# DON'T DO SPACE = ROWS-1 UNLIKE PREVIOUS PYRAMID PROGRAMS
for i in range(rows):
    if space != 0:
        print(" "*space,end=" ")
        for j in range(0,i+1):
            print("*",end=" ")
        print()
        space -= 1
        
    if space == 0:
        rows -= 1
        for i in range(rows):
            print(" "*space,end="  ")
            for j in range (0, back_col):
                print(" *",end="")
            print()
            space += 1
            back_col -= 1

File: Pattern Printing/Javascript/Diamond-pyramid.js

// Printing Diamond pyramid
rows = prompt("Enter the number of of rows you want in a pyramid (It will also give a dropdown pyramid attached to it):");
space = rows;
back_col = rows-1;

for (i=0; i<rows; i++){

    for (_=0; _<space; _++){
        document.write("&nbsp;&nbsp;")
    }

    for (j=0; j<i+1; j++){
        document.write("&nbsp; *")
    }
    document.write("</br>")
    space -= 1

    if (space == 0){
        rows -= 1;
        space += 2

        for (k=0; k<rows; k++){

            for (_=0; _<space; _++){
                document.write("&nbsp;&nbsp;")
            }
            space += 1

            for (l=0; l<back_col; l++){
                document.write("&nbsp; *")
            }
            document.write("</br>")
            back_col -= 1
        }
    }
}